queue/group_remote.go

```diff
@@ -250,21 +261,24 @@ func (g *remoteQueueGroup) Prune(ctx context.Context) error {
 	if err != nil {
 		return errors.Wrap(err, "problem getting collections")
 	}
-	for i, coll := range colls {
+	collsToCheck := []string{}
+	for _, coll := range colls {
 		// This is an optimization. If we've added to the queue recently enough, there's no
 		// need to query its contents, since it cannot be old enough to prune.
 		if t, ok := g.ttlMap[g.idFromCollection(coll)]; ok && time.Since(t) < g.ttl {
-			g.remove(colls, i)
+			// pass
+		} else {
+			collsToCheck = append(collsToCheck, coll)
 		}
 	}
 	catcher := grip.NewBasicCatcher()
 	wg := &sync.WaitGroup{}
-	collsDeleteChan := make(chan string, len(colls))
+	collsDeleteChan := make(chan string, len(collsToCheck))
 	collsDropChan := make(chan string)
 
 	go func() {
 		defer recovery.LogStackTraceAndContinue("panic in pruning collections")
-		for _, coll := range colls {
+		for _, coll := range collsToCheck {
```

Comment:
given that this collsDropChan is buffered, do we need to do this in a thread?

Reply:
done